src/code_transformation/util.jl

```diff
@@ -110,3 +110,52 @@ replace_vararg(typ::SymOrExpr, vararg_info::Tuple) =
         vararg_info[2] == :no_N || vararg_info[2] == :Vararg ?
             replace_body(typ, :(Vararg{$(get_body(typ))})) :
             replace_body(typ, :(Vararg{$(get_body(typ)), $(vararg_info[2])}))
+
+"""
+    parse_kwargs(nt_expr) -> NamedTuple
+
+Accepts an expression containing a `NamedTuple` literal and parses it into a `NamedTuple`
+with expressions as values.
+"""
+function parse_kwargs(nt_expr)
+    if isempty(nt_expr.args)
+        return NamedTuple()
+    end
+
+    first_arg = first(nt_expr.args)
+    if first_arg.head == :parameters
+        return parse_kwargs_parameters(nt_expr)
+    elseif first_arg.head == :(=)
+        return parse_kwargs_tuple(nt_expr)
+    else
+        throw(ArgumentError("Unsupported expression $nt_expr for kwargs;"
+            * " they must be passed as a NamedTuple literal"))
+    end
+end
+
+function parse_kwargs_tuple(tup_expr)
+    nt_names = Tuple(first(ex.args) for ex in tup_expr.args)
+    return NamedTuple{nt_names}(last(ex.args) for ex in tup_expr.args)
+end
+
+function parse_kwargs_parameters(param_tuple_expr)
+    param_expr = param_tuple_expr[1]
+
```

Comment:
Can you now just call  `parse_kwargs_tuple(param_expr)`,
rather than duplicating it's code?

Reply:
I pondered this. It's technically the same code but the expression types are actually different. I can do that but I'll add a comment.